Build failures on 6.19

```
src/modules/net/net.c: In function 'maybe_reconnect':
src/modules/net/net.c:143:45: error: passing argument 2 of 'sk->ops->connect' from incompatible pointer type [-Wincompatible-pointer-types]
  143 |         switch ((err = sk->ops->connect(sk, (struct sockaddr *)&saddr, sizeof(saddr), O_WRONLY))) {
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                                             |
      |                                             struct sockaddr *
src/modules/net/net.c:143:45: note: expected 'struct sockaddr_unsized *' but argument is of type 'struct sockaddr *'
  CC [M]  src/modules/database/arch/x86/p_switch_idt/p_switch_idt.o
make[4]: *** [/usr/src/linux-6.19.0-6.19-alt0.rc3/scripts/Makefile.build:287: src/modules/net/net.o] Error 1
```

--- a/src/modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.c
+++ b/src/modules/exploit_detection/syscalls/pCFI/p___queue_work/p___queue_work.c
@@ -26,7 +26,13 @@ static int p_pcfi___queue_work_entry(struct kprobe *p_ri, struct pt_regs *p_regs
 
    p_ed_pcfi_cpu(0);
 
-   if (!in_irq() && p_is_ed_task(current)) {
+   if (
+#if !defined(in_hardirq) && defined(in_irq)
+       !in_irq()
+#else
+       !in_hardirq()
+#endif
+       && p_is_ed_task(current)) {
       /* Do not take ED lock */
       if ((p_tmp = ed_task_trylock_current())) {
          p_ed_validate_current(p_tmp);

--- a/src/modules/net/net.c
+++ b/src/modules/net/net.c
@@ -140,7 +140,13 @@ static void maybe_reconnect(void)
 	saddr.sin_family = AF_INET;
 	saddr.sin_port = htons(net_server_port);
 	saddr.sin_addr.s_addr = net_server_addr_n;
-	switch ((err = sk->ops->connect(sk, (struct sockaddr *)&saddr, sizeof(saddr), O_WRONLY))) {
+	switch ((err = sk->ops->connect(sk,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+	    (struct sockaddr_unsized *)
+#else
+	    (struct sockaddr *)
+#endif
+	    &saddr, sizeof(saddr), O_WRONLY))) {
 	case 0:
 	case -EINPROGRESS:
 		break;

